### PR TITLE
feat: Support new/instanceof/use trait in `fully_qualified_strict_types`

### LINKED ISSUE #5594: Support for object initialisation in `fully_qualified_strict_types`
## Rule request

`fully_qualified_strict_types` should shorten FQCN to imported symbols in `new \Foo\Bar()`-like cases.

### Code snippet that shows how the rule would fix PHP code

```diff
<?php

use Foo\Bar;
use Some\CustomException;

-$a = new \Foo\Bar();
+$a = new Bar();

-throw new \Some\CustomException('ohoh');
+throw new CustomException('ohoh');
```

### LINKED ISSUE #7568: Support for shortening FQCN for trait usage
```diff
<?php

use Bar\Baz;

class Foo {
-    use \Bar\Baz;
+    use Baz;
}
```

_Originally posted by @mvorisek in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7459#discussion_r1403177320_

### PR DESCRIPTION
**Pull request by mvorisek**
fix #5594
fix #7568

### CODE REVIEW

File: src/Fixer/Import/FullyQualifiedStrictTypesFixer.php [hunk 1]
```diff
@@ -162,12 +162,15 @@ public function getPriority(): int
...
-            T_EXTENDS,
             T_CATCH,
             T_DOUBLE_COLON,
+            T_DOC_COMMENT,
+            T_EXTENDS,
+            T_FUNCTION,
+            T_IMPLEMENTS,
+            T_INSTANCEOF,
+            T_NEW,
+            CT::T_USE_TRAIT,
```
Comment (Wirone) [inline src/Fixer/Import/FullyQualifiedStrictTypesFixer.php:173]:
```suggestion
            CT::T_USE_TRAIT,
            T_CATCH,
            T_DOC_COMMENT,
            T_DOUBLE_COLON,
            T_EXTENDS,
            T_FUNCTION,
            T_IMPLEMENTS,
            T_INSTANCEOF,
            T_NEW,
```
I see you wanted to sort this list, so it here's some minor improvement. I suspect you wanted `CT::T_USE_TRAIT` to be at the end as you treated `CT::` part as a detail and looked only at `T_USE_TRAIT`, but IMHO it should be sorted naturally, with all custom tokens at the beginning.

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php [hunk 2]
```diff
@@ -376,6 +410,8 @@ function A(\Z\B\C\DE\Foo $fix) {}
 namespace Foo\Test;
 use Other\BaseClass;
 use Other\CaughtThrowable;
+use Other\Cl;
+use Other\Cl2;
```
Comment (Wirone) [inline tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:414]: Please use meaningful names like in the other symbol in this test case, like:


```suggestion
use Other\InstantiatedClass;
use Other\InstanceOfClass;
```

It does not matter much, but would be consistent.

**Review by Wirone — CHANGES_REQUESTED** [review]
Great work! Some minor things to sort out.

File: src/Fixer/Import/FullyQualifiedStrictTypesFixer.php (hunk 1 above)
Comment (mvorisek) [reply src/Fixer/Import/FullyQualifiedStrictTypesFixer.php]: done

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php (hunk 2 above)
Comment (mvorisek) [reply tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php]: 👍 done

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you very much @mvorisek 🍻!

**Pull request merged**